src/main/java/uk/gov/laa/ccms/caab/mapper/ClientDetailMapper.java

```diff
@@ -132,4 +132,128 @@ default Date mapDateOfBirth(ClientDetails clientFormData) {
+  @Mapping(target = "homeOfficeNumber",
+      source = "details.personalInformation.homeOfficeNumber")
+  @Mapping(target = "gender",
+      source = "details.personalInformation.gender")
+  @Mapping(target = "maritalStatus",
+      source = "details.personalInformation.maritalStatus")
+  @Mapping(target = "vulnerableClient",
+      source = "details.personalInformation.vulnerableClient")
+  @Mapping(target = "highProfileClient",
+      source = "details.personalInformation.highProfileClient")
+  @Mapping(target = "vexatiousLitigant",
+      source = "details.personalInformation.vexatiousLitigant")
+  @Mapping(target = "mentalIncapacity",
+      source = "details.personalInformation.mentalCapacityInd")
+  @Mapping(target = "telephoneHome",
+      source = "details.contacts.telephoneHome")
+  @Mapping(target = "telephoneWork",
+      source = "details.contacts.telephoneWork")
+  @Mapping(target = "telephoneMobile",
+      source = "details.contacts.mobileNumber")
+  @Mapping(target = "emailAddress",
+      source = "details.contacts.emailAddress")
+  @Mapping(target = "passwordReminder",
+      source = "details.contacts.passwordReminder")
+  @Mapping(target = "correspondenceMethod",
+      source = "details.contacts.correspondenceMethod")
+  @Mapping(target = "correspondenceLanguage",
+      source = "details.contacts.correspondenceLanguage")
+  @Mapping(target = "noFixedAbode",
+      source = "details.noFixedAbode")
+  @Mapping(target = "country",
+      source = "details.address.country")
+  @Mapping(target = "ethnicOrigin",
+      source = "details.ethnicMonitoring")
+  @Mapping(target = "disability",
+      source = "details.disabilityMonitoring.disabilityType",
+      qualifiedByName = "mapListToString")
+  @Mapping(target = "specialConsiderations",
+      source = "details.specialConsiderations")
+  @Mapping(target = "houseNameNumber", source = "details.address.house")
+  @Mapping(target = "addressLine1", source = "details.address.addressLine1")
+  @Mapping(target = "addressLine2", source = "details.address.addressLine2")
+  @Mapping(target = "cityTown", source = "details.address.city")
+  @Mapping(target = "postcode", source = "details.address.postalCode")
+  @Mapping(target = "county", source = "details.address.county")
+  ClientDetails toClientDetails(ClientDetail soaClientDetail);
+
+  /**
+   * Translates a list to a sting, using the first element.
+   *
+   * @param values The list of strings to convert
+   * @return Translated String value.
+   */
+  @Named("mapListToString")
+  default String mapListToString(List<String> values) {
+    if (values != null && !values.isEmpty()) {
+      return values.get(0);
+    }
+    return null;
+  }
+
+  /**
+   * Translates a Date and extracts the day.
+   *
+   * @param date The date to convert
+   * @return Translated String value.
+   */
+  @Named("mapDayFromDate")
+  default String mapDayFromDate(Date date) {
+    if (date != null) {
+      Calendar calendar = Calendar.getInstance();
```

Comment:
Just for the sake of modernising, should this be done using LocalDate rather than calendar?
`dateToConvert.toInstant()
      .atZone(ZoneId.systemDefault())
      .toLocalDateTime();`

Reply:
sorted